app/src/main/java/com/spascoding/taskycourse/MainViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.spascoding.taskycourse
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfo
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfoManager
+import com.spascoding.taskycourse.ui.theme.MainState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private var userInfoManager: UserInfoManager
+): ViewModel() {
+
+    private val _state = mutableStateOf(MainState(
+        userInfo = UserInfo(),
+        isReady = false,
+    ))
+    val state: State<MainState> = _state
+
+    init {
+        viewModelScope.launch {
+            userInfoManager.userInfoFlow.collect { userInfo ->
+                _state.value = state.value.copy(
+                    userInfo = userInfo,
+                    isReady = true,
+                )
```

Comment:
In general, I'd always use `_stateFlow.update { it.copy(...) }` to avoid race conditions. This is how it could happen:

*StateFlow holds value A*
- Thread 1 calls `copy()` on StateFlow with value A
- Before Thread 1 assigns its changed copy, Thread 2 updates StateFlow with value B
- Now, Thread 1 assigns its changed copy, which is still based on value A though, but the latest value is already value B. Therefore, value B would never be considered here.

Not always an issue, but I worked on highly stateful apps with frequently changing states, in which this really bit me, since such bugs are super hard to find.
Using `_state.value = _state.value.copy(...)` is fine if you always call this from a single thread (which you do here). But in ViewModels, you're very likely to not do that as soon as you'd change the state inside a `viewModelScope`.